nir/serialization.py

```diff
@@ -4,46 +4,59 @@
 import h5py
 import numpy as np
 
+import nir.ir as ir
 import nir
 
 
+def _read_metadata(node: Any) -> Dict[str, Any]:
+    if "metadata" in node.keys():
+        return {"metadata": {k: v[()] for k, v in node["metadata"].items()}}
+    else:
+        return {}
+
+
 def try_byte_to_str(a: Union[bytes, Any]) -> Union[str, Any]:
     return a.decode("utf8") if isinstance(a, bytes) else a
 
 
-def read_node(node: Any) -> nir.typing.NIRNode:
+def read_node(node: Any) -> ir.NIRNode:
     """Read a graph from a HDF/conn5 file."""
     if node["type"][()] == b"Affine":
-        return nir.Affine(weight=node["weight"][()], bias=node["bias"][()])
+        return ir.Affine(
```

Comment:
Why the change from nir.Affine to ir.Affine? All Node types are directly accessible as nir.Affine, nir.Conv2d etc.

Reply:
Good catch! It should indeed be accessed via `nir`